x-pack/plugins/osquery/public/agents/use_agent_groups.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useState } from 'react';
+import { useQuery } from 'react-query';
+import { useKibana } from '../common/lib/kibana';
+import { useOsqueryPolicies } from './use_osquery_policies';
+
+import {
+  OsqueryQueries,
+  AgentsRequestOptions,
+  AgentsStrategyResponse,
+} from '../../common/search_strategy';
+
+import { generateTablePaginationOptions } from './helpers';
+import { Overlap } from './types';
+
+interface Group {
+  name: string;
+  size: number;
+}
+
+export const useAgentGroups = () => {
+  const { data } = useKibana().services;
+  const { osqueryPolicies, osqueryPoliciesLoading } = useOsqueryPolicies();
+
+  const [platforms, setPlatforms] = useState<Group[]>([]);
+  const [policyOptions, setPolicyOptions] = useState<Group[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [overlap, setOverlap] = useState<Overlap>(() => ({}));
+  const [totalCount, setTotalCount] = useState<number>(0);
+  useQuery(['agentGroups', osqueryPoliciesLoading], async () => {
```

Comment:
why do we need `osqueryPoliciesLoading` here?

Reply:
`osqueryPolicies` is used to filter the platforms query down to relevant agents. i think it would make sense to throw it into the `enabled` option, now that i think about it.

Reply:
yeah, `enabled` sounds good to me ;)